README.md

```diff
@@ -365,6 +367,69 @@ In order to estimate total records that actually added to the index, you can sen
 `totalRecords`, e.g. `GET /search/instances?query=id="*"`. Alternatively you can query Elasticsearch directly,
 see [ES search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-all-query.html#query-dsl-match-all-query).
 
+## Indexing of Instance Records
+
+The section below describes the data indexing for ```instance```, ```subject```, ```contribution``` and ```classification``` resource types.
+Since ```subject```, ```contribution``` and ```classification``` are derived or child records of an Instance, in short, we are referring to
+indexing of these and ```instance``` as _Indexing of Instances_ (or _Instances Indexing_) and resources as Instance resources.
+
+This is an improved and recommended version of instances indexing than that which is achieved with
+```POST [OKAPI_URL]/search/index/inventory/reindex``` and documented in [Data Indexing](#data-indexing) section
+
+The whole indexing procedure consist of two steps:
+1. Merge - building the necessary data model and collection of data in accordance with the built model
+2. Upload - querying the data and adding them into OpenSearch index
+
+Merge step can be considered as Data Aggregation or Data Reloading. We need to run this step when there is a breaking change introduced to index structure (mapping).
+Normally, we run Instances indexing with this step rarely, as introducing the breaking changes does not happen that frequently. The step aggregates all instances data
+available for the tenant into specific data models and keeps them there. For Create/Update/Delete operations performed on specific Instances mod-search will
+update or sync the aggregated data of those specific Instances, so that the aggregated data is always actual and represents the latest state of an instance resources.
+
+Upload step queries the aggregated data for the specific Instance resources type and then runs the indexing by populating the data into resource's OpenSearch index.
+
+We can execute both Merge and Upload steps with
+```http
+POST /search/index/instance-records/reindex/full
+```
+It runs the async process which performs both steps described above.
+
+If we do not want to run Merge step, but just Upload one then we can call the following async API:
```

Comment:
You should replace "do not want" to something that describes why it could be not required to run merge step and upload is enough, and why it could be required to run merge step.